### PR TITLE
fix/edged and edge controller: make customsiz labels available when restart …

### PR DESCRIPTION
**Pull request by ttlv**
What type of PR is this?
/kind feature

What this PR does / why we need it:
update the labels field in edgecore's file and synchronize the labels to cloud when restart edgecore 
Which issue(s) this PR fixes:

Fixes #
#2764

### CODE REVIEW

File: edge/pkg/edged/edged.go [hunk 1]
```diff
@@ -213,6 +213,7 @@ type edged struct {
 	rootDirectory      string
 	gpuPluginEnabled   bool
 	version            string
+	labels             map[string]string
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:216]: What's this labels used?

Comment (ttlv) [reply edge/pkg/edged/edged.go:216]: Because I don't want to get labels form config yaml again,so I make a field of labels in edged struct.

Comment (fisherxu) [reply edge/pkg/edged/edged.go:216]: Did not find where this label was used..

Comment (ttlv) [reply edge/pkg/edged/edged.go:216]: ```initialNode()``` this function

Comment (fisherxu) [reply edge/pkg/edged/edged.go:216]: Where's e.label used after init in initialNode()?

Comment (ttlv) [reply edge/pkg/edged/edged.go:216]: sorry, this code be omitted and I have added

File: edge/pkg/edged/edged.go [hunk 2]
```diff
@@ -320,6 +321,13 @@ func (e *edged) Start() {
 	// handled by pod workers).
 	go utilwait.Until(e.podKiller.PerformPodKillingWork, 5*time.Second, utilwait.NeverStop)
 
+	// update node label
+	node, _ := e.GetNode()
+	node.Labels = e.labels
+	if err := e.metaClient.Nodes(e.namespace).Update(node); err != nil {
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go:327]: I'm afraid whether this updating will overide other lables setted from cloud like kubectl, would you like to have a test?

File: edge/pkg/edged/edged.go (hunk 1 above)
Comment (fisherxu) [reply edge/pkg/edged/edged.go:216]: Sounds good.

File: edge/pkg/edged/edged.go (hunk 2 above)
Comment (ttlv) [reply edge/pkg/edged/edged.go:327]: thers is a problem, if we define customsiz labels like ```key1=value1``` in edgecore's configfile and we use kubectl label commonand to update label like ```key1=value2```,if we restart edgecore, the value of key1 label will become value1 again

Comment (fisherxu) [reply edge/pkg/edged/edged.go:327]: Yes, and here we should use patch to update the node spec(maybe can do it in cloudcore).

File: cloud/pkg/edgecontroller/controller/upstream.go [hunk 3]
```diff
@@ -698,6 +698,10 @@ func (uc *UpstreamController) updateNode() {
 					klog.Warningf("message: %s process failure with error: %s, name: %s", msg.GetID(), err, name)
 					continue
 				}
+				// update node labels
+				for key, value := range noderequest.Labels {
+					getNode.Labels[key] = value
+				}
```
Comment (fisherxu) [inline cloud/pkg/edgecontroller/controller/upstream.go:704]: Patch in line 713

File: edge/pkg/edged/edged.go (hunk 2 above)
Comment (ttlv) [reply edge/pkg/edged/edged.go:327]: sounds good,I have used patch to update node object

**Review by fisherxu — COMMENTED** [review]
Looks good, has any test?

**Comment by ttlv** [comment]
> Looks good, has any test?

you mean write some test cases or test manually，I have tested manually in my kubeedge cluster

**Comment by fisherxu** [comment]
> > Looks good, has any test?
> 
> you mean write some test cases in e2e or test manually，I have tested manually in my kubeedge cluster

Okey, please fix all the failed ci.

**Comment by ttlv** [comment]
> > > Looks good, has any test?
> > 
> > 
> > you mean write some test cases in e2e or test manually，I have tested manually in my kubeedge cluster
> 
> Okey, please fix all the failed ci.

ok failed ci has been fixed and ready to get lgtm :)

**Comment by fisherxu** [comment]
Also need to squash the commits :) @ttlv

**Comment by ttlv** [comment]
> Also need to squash the commits :) @ttlv

yeap,I have squashed all the commits into one commit :)

**Review by fisherxu — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**